### PR TITLE
Fix several deprecations

### PR DESCRIPTION
**Pull request by TacoTheDank**
### 1st commit
- Update gradle to 7.0.2 using git bash
- Update RecyclerView 1.2.0 -> 1.2.1 ([changelog](https://developer.android.com/jetpack/androidx/releases/recyclerview#version_121_3))

### 2nd commit
- Utilize FragmentContainerView (see [here](https://developer.android.com/jetpack/androidx/releases/fragment#1.2.0))

### 3rd commit
- See [this](https://developer.android.com/jetpack/androidx/releases/fragment#1.3.0-alpha04) for this and the next commit...
- Fix onActivityResult deprecations
- Use ContextCompat.startForegroundService [here](https://github.com/emanuele-f/PCAPdroid/commit/56f366955d032d6106f08cbb6e732d4789a9cf21#diff-1d2191f73ed5aaefe7b1e0b8a74e87054c2ff86a10bbed571a2f40626922b551R487)

### 4th commit
- Fix onRequestPermissionsResult deprecation

### 5th commit
- Rename MyStateAdapter to something more relevant
- Clean up the state adapter a bit (keeps more of the state adapter stuff under the one inner class)

Again, I was unable to test these changes myself due to building errors (that I cannot resolve due to lack of hard drive space), but I'm rather confident that these changes work.

### CODE REVIEW

**Comment by emanuele-f** [comment]
Thanks, I will review the changes soon

File: app/src/main/res/layout/settings_activity.xml [hunk 1]
```diff
@@ -3,11 +3,10 @@
     android:layout_height="match_parent"
     android:orientation="vertical">
 
-    <FrameLayout
-        android:id="@+id/settings"
+    <androidx.fragment.app.FragmentContainerView
+        android:id="@+id/settings_container"
         android:layout_width="match_parent"
-        android:layout_height="0dp"
-        android:layout_weight="1" />
```
Comment (emanuele-f) [inline app/src/main/res/layout/settings_activity.xml:10]: `match_parent` is not equivalent to the removed lines and it would hide the adContainer below

**Review by emanuele-f — CHANGES_REQUESTED** [review]
Please check my comment below and test your changes before submitting. Testing is essential for any PR and it will improve the quality of your contributions. Common tests include rotating the device screen and putting the app in background. You should assume that any untested line does not work, as software is often infamous and bugs can hide even in 1 line change

**Comment by TacoTheDank** [comment]
@emanuele-f Sorry about that. In any other cases, I always test my PRs before I make them to iron out any bugs. However, here's an example of why I can't seem to build PCAPdroid (some stuff x'ed out for privacy):

```
> Task :app:externalNativeBuildDebug FAILED
Build multiple targets ndpi_armeabi-v7a vpnproxy-jni_armeabi-v7a libpcapd.so_armeabi-v7a
ninja: Entering directory `C:\Users\xx\xx\xx\xx\PCAPdroid\app\.cxx\cmake\debug\armeabi-v7a'

Execution failed for task ':app:externalNativeBuildDebug'.
> Build command failed.
  Error while executing process C:\Users\xx\AppData\Local\Android\Sdk\cmake\3.18.1\bin\ninja.exe with arguments {-C C:\Users\xx\xx\xx\xx\PCAPdroid\app\.cxx\cmake\debug\armeabi-v7a libpcapd.so ndpi vpnproxy-jni}
  ninja: Entering directory `C:\Users\xx\xx\xx\xx\PCAPdroid\app\.cxx\cmake\debug\armeabi-v7a'
  [1/215] Generating C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/libpcap/grammar.c, C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/libpcap/scanner.c
  FAILED: C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/libpcap/grammar.c C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/libpcap/scanner.c 
  cmd.exe /C "cd /D C:\Users\xx\xx\xx\xx\PCAPdroid\app\.cxx\cmake\debug\armeabi-v7a\pcapd && cd C:/Users/xx/xx/xx/xx/PCAPdroid/app/src/main/jni/../../../../submodules/libpcap && ./configure && make grammar.c scanner.c"
  '.' is not recognized as an internal or external command,
  operable program or batch file.
  [2/215] Generating C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/nDPI/src/include/ndpi_api.h
  FAILED: C:/Users/xx/xx/xx/xx/PCAPdroid/submodules/nDPI/src/include/ndpi_api.h 
  cmd.exe /C "cd /D C:\Users\xx\xx\xx\xx\PCAPdroid\app\.cxx\cmake\debug\armeabi-v7a\vpnproxy-jni && cd C:/Users/xx/xx/xx/xx/PCAPdroid/app/src/main/jni/../../../../submodules/nDPI && ./autogen.sh && ./configure --disable-gcrypt"
  '.' is not recognized as an internal or external command,
  operable program or batch file.
  [3/215] Building C object common/CMakeFiles/common.dir/utils.c.o
  [4/215] Building C object common/CMakeFiles/common.dir/uid_resolver.c.o
  [5/215] Building C object common/CMakeFiles/common.dir/uid_lru.c.o
  [6/215] Building C object zdtun_build/CMakeFiles/zdtun.dir/utils.c.o
  [7/215] Building C object zdtun_build/CMakeFiles/zdtun.dir/zdtun.c.o
  ninja: build stopped: subcommand failed.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

I follow the build cloning instructions and everything, so I don't know why this is happening...

**Comment by emanuele-f** [comment]
PCAPdroid is intended to be built on a linux distribution, for example Debian or Ubuntu. I've updated the docs to make this clear. You may find workarounds for Windows (e.g. MinGW) but doing this is a pain, much better to built it into an ubuntu vm (e.g. via Virtualbox).

**Comment by TacoTheDank** [comment]
@emanuele-f I see, thanks!

**Pull request merged**